flocx_market/matcher/matcher.py

```diff
@@ -89,12 +89,34 @@ def apply_operator(val1, val2, op):
 
 def match_specs(match_expression, data):
     for exp in match_expression:
-        if None in exp:
-            return False
 
         j_val = jmespath.search(exp[0], data)
         op = exp[1]
         val = exp[2]
+
         if not apply_operator(j_val, val, op):
             return False
+
     return True
+
+
+def get_all_matching_offers(specs,
+                            start_time=None,
+                            end_time=None,
+                            first=False):
```

Comment:
We don't seem to be using the `first` parameter anywhere in the code right now.

Reply:
Yes, I just kept it so that if someome wants just one matching offer then instead going through all and return a list we can return one. But I can remove it